crud/src/main/java/com/redhat/lightblue/crud/mongo/Translator.java

```diff
@@ -679,7 +679,9 @@ private DBObject translateNaryFieldRelationalExpression(FieldTreeNode context, N
     }
 
     private DBObject translateUnaryLogicalExpression(FieldTreeNode context, UnaryLogicalExpression expr) {
-        return new BasicDBObject(UNARY_LOGICAL_OPERATOR_MAP.get(expr.getOp()), translate(context, expr.getQuery()));
+        List<DBObject> l=new ArrayList<>(1);
+        l.add(translate(context,expr.getQuery()));
+        return new BasicDBObject("$nor", l);
```

Comment:
Fixed.

On Tue, Mar 10, 2015 at 6:57 AM, Dennis Crissman <notifications@github.com>
wrote:

> In crud/src/main/java/com/redhat/lightblue/crud/mongo/Translator.java
> <https://github.com/lightblue-platform/lightblue-mongo/pull/80#discussion_r26118547>
> :
>
> > @@ -679,7 +679,9 @@ private DBObject translateNaryFieldRelationalExpression(FieldTreeNode context, N
> >      }
> >
> >      private DBObject translateUnaryLogicalExpression(FieldTreeNode context, UnaryLogicalExpression expr) {
> > -        return new BasicDBObject(UNARY_LOGICAL_OPERATOR_MAP.get(expr.getOp()), translate(context, expr.getQuery()));
> > +        List<DBObject> l=new ArrayList<>(1);
> > +        l.add(translate(context,expr.getQuery()));
> > +        return new BasicDBObject("$nor", l);
>
> Hard coding "$nor" instead of using the UNARY_LOGICAL_OPERATOR_MAP might
> be confusing to someone attempting to debug or change the value later.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-mongo/pull/80/files#r26118547>
> .
>